### PR TITLE
implemented tree endpoints  for getting our card

### PR DESCRIPTION
**Pull request by DanielSzczucki**

### CODE REVIEW

File: routers/fischcardRouter.ts [hunk 1]
```diff
@@ -37,35 +41,161 @@ fischcardRouter
...
       }
     } catch (error) {
       res.status(500).json({
-        message: "Something went wrong, please try again later",
+        message: error.message,
+        cards: null,
+      });
+    }
+  })
+  .delete("/card/:id", async (req, res) => {
```
Comment (Sumick) [inline routers/fischcardRouter.ts:166]: **nitpick:** This should not be a part of this pull request

File: routers/fischcardRouter.ts [hunk 2]
```diff
@@ -37,35 +41,161 @@ fischcardRouter
...
+        });
+      }
+    } catch (error) {
+      res.status(500).json({
+        message: error.message,
+      });
+    }
+  })
+
+  .get("/cards", async (req, res) => {
```
Comment (Sumick) [inline routers/fischcardRouter.ts:84]: **issue:** It should not be such a big function. Router should be responsible for routing, not for handling business logic. We should split it into smaller pieces / functions, and then import them to the router. I will explain in detail in review summary

**Review by Sumick — CHANGES_REQUESTED** [review]
Not bad, but we need to improve maintainability of that solution. First thing is that you should not keep all your business logic inside the router. It's hard to debug, read - and router is not for that purpose. I suggest you to create for example `services/fischcard.service.ts` where you will keep all your functions like `findAllById`, `updateOne` etc. Your service should be responsible for such logic. Then you should import such methods into your router.

To summarize, two things to improve:

- [ ] Split this router into smaller pieces, and separate business logic
- [ ] Remove all changes that are not within this task scope (like deleting)

**Comment by DanielSzczucki** [comment]
not added, need to fix it a little bit
